lib/tests/test_conflicts.rs

```diff
@@ -1065,6 +1189,249 @@ fn test_update_conflict_from_content_simplified_conflict() {
     );
 }
 
+#[test]
+fn test_two_sided_conflict_with_different_markers() {
+    let test_repo = TestRepo::init();
+    let store = test_repo.repo.store();
+
+    let path = RepoPath::from_internal_string("file");
+    let base_id = testutils::write_file(
+        store,
+        path,
+        indoc! {"
+            line 1
+            line 2 base
+            line 3 base
+            line 4 base
+            line 5
+        "},
+    );
+    let a_id = testutils::write_file(
+        store,
+        path,
+        indoc! {"
+            line 1
+            line 2 a.1
+            line 3 a.2
+            line 4 base
+            line 5
+        "},
+    );
+    let b_id = testutils::write_file(
+        store,
+        path,
+        indoc! {"
+            line 1
+            line 2 b.1
+            line 3 base
+            line 4 b.2
+            line 5
+        "},
+    );
+
+    let conflict = Merge::from_removes_adds(
+        vec![Some(base_id.clone())],
+        vec![Some(a_id.clone()), Some(b_id.clone())],
+    );
+
+    // Make sure each materialized conflict is able to be parsed correctly
+    let materialize_and_verify_parsed = |conflict_marker_style| {
+        let content = materialize_conflict_string(store, path, &conflict, conflict_marker_style);
+        // Validate parsing conflict markers with different config options
+        for other_style in [ConflictMarkerStyle::Diff, ConflictMarkerStyle::Snapshot] {
+            let parsed =
+                update_from_content(&conflict, store, path, content.as_bytes(), other_style)
+                    .block_on()
+                    .unwrap();
+
+            assert_eq!(
+                parsed, conflict,
+                "parse {conflict_marker_style:?} conflict markers with {other_style:?}"
+            );
+        }
+
+        content
+    };
```

Comment:
It looks like we update from an unmodified conflict here, right? It looks like it's only the format that changed. If that's right, I think it would be clearer if we made one of two changes:

1. Pass in a different conflict on line 1239. It could be a completely different conflict as long as it has the right arity.
2. Use `materialize_conflict_string` to produce the materialized content. The test could then become a double loop like this:
```rs
for old_style in [ConflictMarkerStyle::Diff, ConflictMarkerStyle::Snapshot] {
  let content = materialize_conflict_string(store, path, &conflict, conflict_marker_style);
  for new_style in [ConflictMarkerStyle::Diff, ConflictMarkerStyle::Snapshot] {
    let parsed = ...;
    assert_eq!(...);
  }
}
```

I'd probably go with 2 because it's simpler.

The same applies to the next test case.

Does that make sense?

Reply:
I was originally trying to do two different things in the test: checking that it materialized in the correct format, and checking that parsing the materialized conflict works regardless of the configured conflict marker style. I agree that it's a bit unclear, so I converted that test to your "option 2", and I added separate tests for the format of materialized conflicts. I think it's a bit cleaner, thanks!